### PR TITLE
🚀 Add advanced string, collection, and StringBuilder extension methods

### PR DESCRIPTION
**Pull request by stescobedo92**
This update introduces new functionalities and improvements to the BinaryCoffee.Extensions library, along with comprehensive unit tests for all methods.

Features Added:
String Extensions:

ToSlug: Convert strings into URL-friendly slugs.
StripTags: Remove HTML or XML tags from strings.
Reverse: Reverse the content of strings.
CapitalizeWords: Capitalize the first letter of each word in strings.
NormalizeSpaces: Trim and normalize spaces in strings.
MatchesPattern: Validate strings against regular expression patterns.
RemoveCharacters: Remove specific characters from strings.
Collection Extensions:

TransformAsync: Apply asynchronous transformations to collection elements.
AddNested: Simplify working with nested collections in dictionaries.
StringBuilder Extensions:

AppendRepeated: Append a value to a StringBuilder multiple times.
Additional Changes:
Enhanced test coverage for all methods, including edge cases (null, empty strings).
Comprehensive documentation in Markdown for the library, featuring examples and usage guidelines.

### CODE REVIEW

File: BinaryCoffee.Extensions/Helpers/AsyncExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+            {
+                builder.Append(await formatter(item));
+                builder.Append(separator);
+            }
+
+            return builder.ToString().TrimEnd(separator.ToCharArray());
+        }
+
+        /// <summary>
+        /// Converts a string to CamelCase format asynchronously.
```
Comment (pedro-gilmora) [inline BinaryCoffee.Extensions/Helpers/AsyncExtensions.cs:33]: I see no reason for this. JoinAsync would be very useful, but non of these below would regular cases

File: BinaryCoffee.Extensions/Helpers/GeneralExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    /// <summary>
+    /// Adds a value item to a nested collection within a dictionary. If the key does not exist, a new collection is created.
+    /// </summary>
+    /// <typeparam name="TList">The type of the collection that holds the value items.</typeparam>
+    /// <typeparam name="TKey">The type of the key in the dictionary.</typeparam>
+    /// <typeparam name="TValueItem">The type of the value item to be added to the collection.</typeparam>
+    /// <param name="listHash">The dictionary containing the nested collections.</param>
+    /// <param name="key">The key associated with the collection to which the value item will be added.</param>
+    /// <param name="valueItem">The value item to add to the collection.</param>
+    public static void AddNested<TList, TKey, TValueItem>(this Dictionary<TKey, TList> listHash, TKey key, TValueItem valueItem) where TList : ICollection<TValueItem>, new()
```
Comment (pedro-gilmora) [inline BinaryCoffee.Extensions/Helpers/GeneralExtensions.cs:21]: Innecesario. ya existe en CollectionExtensions.cs

File: BinaryCoffee.Extensions/Helpers/StringBuilderExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+/// Inserts a formatted collection of strings into the StringBuilder at the specified index.
+/// </summary>
+/// <typeparam name="T">The type of the elements in the collection.</typeparam>
+/// <param name="builder">The StringBuilder to insert into.</param>
+/// <param name="startIndex">The starting index to insert at.</param>
+/// <param name="strs">The collection of strings to insert.</param>
+/// <param name="formatter">The function to format each element.</param>
+/// <param name="separator">The separator to use between elements (optional).</param>
+/// <returns>The updated StringBuilder.</returns>
+public static StringBuilder InsertJoin<T>(this StringBuilder builder, int startIndex, IEnumerable<T> strs, Func<T, string?> formatter, string separator = "") =>
```
Comment (pedro-gilmora) [inline BinaryCoffee.Extensions/Helpers/StringBuilderExtensions.cs:51]: Enhance null handling

File: BinaryCoffee.Extensions/Helpers/StringBuilderExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+/// <summary>
+/// Inserts a collection of strings into the StringBuilder at the specified index.
+/// </summary>
+/// <typeparam name="T">The type of the elements in the collection.</typeparam>
+/// <param name="builder">The StringBuilder to insert into.</param>
+/// <param name="startIndex">The starting index to insert at.</param>
+/// <param name="strs">The collection of strings to insert.</param>
+/// <param name="separator">The separator to use between elements (optional).</param>
+/// <returns>The updated StringBuilder.</returns>
+public static StringBuilder InsertJoin<T>(this StringBuilder builder, int startIndex, IEnumerable<T> strs, string separator = "") =>
```
Comment (pedro-gilmora) [inline BinaryCoffee.Extensions/Helpers/StringBuilderExtensions.cs:63]: Enhance null handling

**Review by pedro-gilmora — APPROVED** [review]
I'll leave these comments for next release
